chainer/functions/connection/deformable_convolution_2d_sampler.py

```diff
@@ -0,0 +1,139 @@
+from chainer.functions.math.matmul import matmul
+
+
+def deformable_convolution_2d_sampler(x, offset, W, b=None, stride=1, pad=0):
+    """Two-dimensional deformable convolution function using computed offset.
+
+    This is implementation of two-dimensional deformable convolution used in
+    `Deformable Convolutional Networks <https://arxiv.org/abs/1703.06211>`_.
+
+    It takes four variables: the input image ``x``, the offset image
+    ``offset``, the filter weight ``W``, and the bias vector ``b``.
+
+    Notation: here is a notation for dimensionalities.
+
+    - :math:`n` is the batch size.
+    - :math:`c_I` and :math:`c_O` are the number of the input and output,
+      respectively.
+    - :math:`h` and :math:`w` are the height and width of the input image,
+      respectively.
+    - :math:`k_H` and :math:`k_W` are the height and width of the filters,
+      respectively.
+    - :math:`s_Y` and :math:`s_X` are the stride of the filter.
+    - :math:`p_H` and :math:`p_W` are the spatial padding size.
+
+    The output size :math:`(h_O, w_O)` is determined by the following
+    equations:
+
+    .. math::
+
+       h_O &= (h + 2p_H - k_H) / s_Y + 1,\\\\
+       w_O &= (w + 2p_W - k_W) / s_X + 1.
+
+    Args:
+        x (~chainer.Variable): Input variable of shape :math:`(n, c_I, h, w)`.
+        offset (~chainer.Variable): Offset variable of shape
+            :math:`(n, 2 \\cdot k_H \\cdot k_W, h_O, w_O)`. The first
+            :math:`k_H \\cdot k_W` index of the second axis corresponds to
+            the offsets in the horizontal direction. The last
+            :math:`k_H \\cdot k_W` index of the second axis corresponds to
+            the offsets in the vertical direction.
+        W (~chainer.Variable): Weight variable of shape
+            :math:`(c_O, c_I, k_H, k_W)`.
+        b (~chainer.Variable): Bias variable of length :math:`c_O` (optional).
+        stride (int or pair of ints): Stride of filter applications.
+            ``stride=s`` and ``stride=(s, s)`` are equivalent.
+        pad (int or pair of ints): Spatial padding width for input arrays.
+            ``pad=p`` and ``pad=(p, p)`` are equivalent.
+
+    Returns:
+        ~chainer.Variable: Output variable.
+
+    Deformable convolution adds 2D offsets to the regular grid sampling
+    locations in the standard convolution. It enables free form deformation of
+    the sampling grid.
+
+    See `Jifeng Dai, Haozhi Qi, Yuwen Xiong, Yi Li, Guodong Zhang, Han Hu, \
+        Yichen Wei. Deformable Convolutional Networks\
+        <https://arxiv.org/abs/1703.06211>`_
+
+    If the bias vector is given, then it is added to all spatial locations of
+    the output of convolution.
+
+    .. seealso:: :class:`~chainer.links.DeformableConvolution2D`
+
+    .. admonition:: Example
+
+        >>> x = np.random.uniform(0, 1, (2, 3, 4, 7)).astype(np.float32)
+        >>> offset = np.random.uniform(
+        ...     0, 1, (2, 2 * 1 * 1, 2, 5)).astype(np.float32)
```

Comment:
The kernel size seems to be `(3, 3)` from L79, so that `kh = 3, kw = 3`. But the offset is initialized with the array which has the shape of `(2, 2 * 1 * 1, 2, 5)`, while the second dimension `2 * 1 * 1` should mean `(2 * k_H * k_W)` from L45. Why isn't that `2 * 3 * 3`?

Reply:
Sorry. The function did not raise error in this erroneous input because unintended broadcasting occurred inside the function.
I fixed the doc.

Reply:
Oh. My bad.